status/status.go

```diff
@@ -0,0 +1,91 @@
+package status
+
+import "sync"
+
+// Status represents a status update from an action within Empire.
+type Status interface {
+	// Returns a human readable string about the status update.
+	String() string
+}
+
+// status implements the Status interface
+type status struct {
+	message string
+}
+
+func (s *status) String() string {
+	return s.message
+}
+
+// NewStatus returns the default implementation of the Status interface
+func NewStatus(m string) *status {
+	return &status{message: m}
+}
+
+// StatusStream is an interface for publishing status updates while executing
+// an Empire action.
+type StatusStream interface {
+	// Publish publishes an update to the status stream
+	Publish(Status)
+
+	// Done finalizes the status stream
+	Done(error)
+}
+
+type SubscribableStream interface {
```

Comment:
> also, i'm not a fan of throwing: panic("Publish called on finalized stream"). given these are just status updates, it feels aggressive to throw a panic. what would be the proper way to warn instead? i couldn't find any examples in the code.

Agreed. If we change the interface to `Publish(ctx, Status)`, then you can do a `logger.Warn(ctx, "message")`.